File: frontend/src/types/report.ts
export interface ReportSummary { sales_total: string; transaction_count: number; items_sold: number; gross_profit: string; profit_complete: boolean; operating_expenses:string; net_profit:string|null; net_profit_complete:boolean; average_transaction_value: string; total_active_products: number; total_units_in_stock: number; low_stock_count: number; out_of_stock_count: number }
export interface TrendPoint { date: string; sales: string; transactions: number; items_sold: number; gross_profit:string; expenses:string; net_profit:string|null; profit_complete:boolean }
export interface TopProduct { product_id: string | null; product_name: string; sku: string; quantity_sold: number; revenue: string }
export interface StockProduct { product_id: string; product_name: string; sku: string; current_stock: number; minimum_stock: number }
export interface InventoryStatus { total_active_products: number; total_units_in_stock: number; low_stock: StockProduct[]; out_of_stock: StockProduct[] }

File: backend/app/schemas/report.py
from datetime import date
from decimal import Decimal
from uuid import UUID

from pydantic import BaseModel


class ReportSummary(BaseModel):
    sales_total: Decimal
    transaction_count: int
    items_sold: int
    gross_profit: Decimal
    profit_complete: bool
    average_transaction_value: Decimal
    total_active_products: int
    total_units_in_stock: int
    low_stock_count: int
    out_of_stock_count: int


class SalesTrendPoint(BaseModel):
    date: date
    sales: Decimal
    transactions: int
    items_sold: int


class TopProduct(BaseModel):
    product_id: UUID | None
    product_name: str
    sku: str
    quantity_sold: int
    revenue: Decimal


class StockProduct(BaseModel):
    product_id: UUID
    product_name: str
    sku: str
    current_stock: int
    minimum_stock: int


class InventoryStatus(BaseModel):
    total_active_products: int
    total_units_in_stock: int
    low_stock: list[StockProduct]
    out_of_stock: list[StockProduct]
